src/datatype.c

```diff
@@ -598,7 +598,7 @@ JL_DLLEXPORT jl_value_t *jl_new_bits(jl_value_t *dt, void *data)
     case  1: *(uint8_t*) v = *(uint8_t*)data;    break;
     case  2: *(uint16_t*)v = jl_load_unaligned_i16(data);   break;
     case  4: *(uint32_t*)v = jl_load_unaligned_i32(data);   break;
```

Comment:
should this be `jl_store_unaligned_i32` or are we guaranteed that i32 is always aligned?

Reply:
We generally guarantee at least pointer alignment for objects that we allocate. The biggest reason all of this is necessary is flisp, which doens't allocate wider than that. In julia, we generally try to give proper alignment. Now that i think about it, I probably just need to set MAX_ALIGN to 8 for wasm to at least some of these fixed. The flisp ones do still end up unaligned though.

Reply:
flisp allocates in pointer-sized units. Do you see anything aligned less than that?

Reply:
No, but the wasm C ABI assumes 8 byte alignment for allocations of that size, but has 32bit pointers (in currently supported modes). There's also a couple places that read from the flisp system image that are even more underaligned, but that's mostly separate.